lib/optional.h

```diff
@@ -0,0 +1,20 @@
+#ifndef LIB_OPTIONAL_H
+#define LIB_OPTIONAL_H
+
+namespace util {
+
+template <typename T>
+struct Optional {
+  bool isNull;
+  T data;
+
+  Optional() : isNull{true} {}
+  Optional(const T& data) : isNull{false}, data{data} {}
```

Comment:
consider overriding the = operator so that when the optional is set equal to something it also is set to isNull=false

Reply:
good idea

Reply:
Actually im not sure if we need to overload = for copy/move assignment. If the Optional is set equal to something, we'll need to check whether that something has a non null field before setting isNull = false. And I think the behavior ends up being the same as the default bytewise copy.

Reply:
In the case that we set an optional equal to another optional I totally agree. However, it may be the case that we wish to set the optional data with just Optional<type> op = type, which is the case I'm referring to. Do you think this case could be useful to cover?

Reply:
Yeah I think that would be useful. Didn't know that you could overload = with another type!